share/lib/python/neuron/rxd/reaction.py

```diff
@@ -133,10 +133,17 @@ def _update_rates(self):
         if ecs_region:
             self._rate_ecs, self._involved_species_ecs = rxdmath._compile(rate, ecs_region)
         
-        #TODO ask about this -> Grab region from regions species in _dest are defined on. I think if we have a species defined on dest its sources must also be defined on that region
-        for sptr in self._dests:
-            s = sptr() if isinstance(sptr(), species.Species) else sptr()._species()
-            self._react_regions = [reg for reg in s._regions]
+        # if a region is specified -- use it
+        if self._regions and self._regions != [None]:
+            self._react_regions = self._regions
+        else:
+        # else find the common regions shared by all sources and  destinations
+            self._react_regions = []
+            regs = []
+            for sptr in self._sources + self._dests:
```

Comment:
Is `for sptr in itertools.chain(self._sources, self._dests)` valid @ferdonline ?